components/server/src/auth/host-context-provider-impl.ts

```diff
@@ -28,17 +28,30 @@ export class HostContextProviderImpl implements HostContextProvider {
     @inject(HostContextProviderFactory)
     protected factory: HostContextProviderFactory;
 
+    async init() {
+        await this.ensureInitialized();
+    }
+
     protected initialized = false;
-    protected ensureInitialized() {
+    protected async ensureInitialized() {
         if (this.initialized) {
             return;
         }
-        this.createFixedHosts()
-        this.updateDynamicHosts();
+        this.createFixedHosts();
+
+        try {
+            await this.updateDynamicHosts();
+        } catch (error) {
+            log.error(`Failed to update dynamic hosts.`, error);
```

Comment:
What happens in this case? Do all dynamic hosts not work in this server instance? Can we recover from this?